### PR TITLE
HHH-10222 - AttributeConverter not applied to @ElementCollection

### PR DESCRIPTION
**Pull request by perceptron8**
Check `CollectionElementConversionTest` first, please. Without my fix, converter would be ignored and there would be attempt to store map's keys in binary form (failing as `Color` is not `Serializable`). With fix, keys are correctly stored as text (converter is used).

Only two conditions were changed in `SimpleValueBinder`.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java [hunk 1]
```diff
@@ -323,12 +323,12 @@ private void applyAttributeConverter(XProperty property, AttributeConverterDescr
...
 		}
 
-		if ( property.isAnnotationPresent( Temporal.class ) ) {
+		if ( !key && property.isAnnotationPresent( Temporal.class ) ) {
 			LOG.debugf( "Skipping AttributeConverter checks for Temporal attribute [%s]", property.getName() );
 			return;
 		}
 
-		if ( property.isAnnotationPresent( Enumerated.class ) ) {
+		if ( !key && property.isAnnotationPresent( Enumerated.class ) ) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java]: For a map key this should still check for the presence of `MapKeyEnumerated`

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java]:
````
if ( ( !key && property.isAnnotationPresent( Enumerated.class ) )
        || key && property.isAnnotationPresent( MapKeyEnumerated.class ) ) ) {
   LOG.debugf( "Skipping AttributeConverter...
}

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java]: I'll make that change as I apply this...

**Comment by sebersole** [comment]
Applied upstream.  Thanks Michal!

File: hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java [hunk 2]
```diff
@@ -323,12 +323,12 @@ private void applyAttributeConverter(XProperty property, AttributeConverterDescr
 			return;
 		}
 
-		if ( property.isAnnotationPresent( Temporal.class ) ) {
+		if ( !key && property.isAnnotationPresent( Temporal.class ) ) {
```
Comment (perceptron8) [inline hibernate-core/src/main/java/org/hibernate/cfg/annotations/SimpleValueBinder.java]: Remember about @MapKeyTemporal too!

**Comment by perceptron8** [comment]
Thanks for merge, Steve! It's very nice that you noticed another missing part BTW. At least in one place, `@MapKeyTemporal` is still missing, but that's nobody's fault but mine.

I think it would be good to to include another map in test, some `Map<Color, Date>` annotated with `@ElementCollection` and `@Temporal `. Once again, I should do that in the first place.

Could you, please?

**Comment by perceptron8** [comment]
Ok! I just checked your last commit, all checks are in place.
Thanks again!

**Pull request closed without merging**